frontend/src/app/main/ui/workspace/sidebar/shortcuts.cljs

```diff
@@ -0,0 +1,294 @@
+   [app.main.data.shortcuts :as ds]
+   [app.main.data.viewer.shortcuts]
+   [app.main.data.workspace :as dw]
+   [app.main.data.workspace.path.shortcuts]
+   [app.main.data.workspace.shortcuts]
+   [app.main.store :as st]
+   [app.main.ui.icons :as i]
+   [app.util.dom :as dom]
+   [app.util.i18n :refer [tr]]
+   [app.util.strings :refer [matches-search]]
+   [clojure.string]
+   [cuerdas.core :as str]
+   [rumext.alpha :as mf]))
+
+
+(defn shortcuts->subsections
+  "A function to obtain the list of subsections and their 
+   associated shortcus from the general map of shortcuts"
+  [shortcuts]
+  (let [subsections (into #{} (mapcat :subsections) (vals shortcuts))
+        get-sc-by-subsection
+        (fn [subsection [k v]]
+          (when (some #(= subsection %) (:subsections v)) k))
+        reduce-sc
+        (fn [acc subsection]
+          (let [shortcuts-by-subsection (keep (partial get-sc-by-subsection subsection) shortcuts)]
+            (assoc acc subsection shortcuts-by-subsection)))]
+    (reduce reduce-sc {} subsections)))
+
+(mf/defc shortcuts-keys
+  [{:keys [content command] :as props}]
+  (let [managed-list (if (coll? content)
+                       content
+                       (conj () content))
+        split-sc     (fn [sc]
+                       (let [sc (cond-> sc (str/includes? sc "++")
+                                        (str/replace "++" "+plus"))]
+                         (if (= (count sc) 1)
+                           [sc]
+                           (str/split sc #"\+| "))))
+        keys-list    (map split-sc managed-list)]
+    [:span.keys
+     (for [keys keys-list]
+       (let [last-element (last keys-list)]
+         [:*
+          (for [key keys]
+            (let [modified-keys {:up    ds/up-arrow
+                                 :down  ds/down-arrow
+                                 :left  ds/left-arrow
+                                 :right ds/right-arrow
+                                 :plus "+"}
+                  macos-keys    {:command "\u2318"
+                                 :option  "\u2325"
+                                 :alt     "\u2325"
+                                 :delete  "\u232B"
+                                 :del     "\u232B"
+                                 :shift   "\u21E7"
+                                 :control "\u2303"
+                                 :esc     "\u238B"
+                                 :enter   "\u23CE"}
+                  is-macos?     (cf/check-platform? :macos)
+                  key           (if (contains? modified-keys (keyword key)) ((keyword key) modified-keys) key)
+                  key           (if (and is-macos? (contains? macos-keys (keyword key))) ((keyword key) macos-keys) key)
+                  unique-key    (str (d/name command) "-" key)]
+              [:*
+               [:span.key-box {:key unique-key} key]]))
+          (when (not= keys last-element) [:span.space ","])]))]))
+
+(mf/defc shotcut-row-generator
```

Comment:
I find it curious the "generator" prefix. This is a function that generates some bunch of html from some parameters. I don't see any special mechanism, different of any other react component. Could it simply be `shortcut-row`?

Reply:
also in section-generator and subsection-generator